test/awss3.jl

```diff
@@ -205,4 +205,25 @@ function awss3_tests(config)
             @test_throws AWS.AWSException s3_delete_bucket(config, bucket_name)
         end
     end
+
+    # these tests are needed because lack of functionality of the underlying AWS API makes certain
+    # seemingly inane tasks incredibly tricky: for example checking if an "object" (file or
+    # directory) exists is very subtle
+    @testset "path naming edge cases" begin
+        s3_create_bucket("name-tests")
```

Comment:
do we need a new bucket here? we might need to think about naming, the bucket we've been using it
```julia
bucket_name = "ocaws.jl.test." * lowercase(Dates.format(now(Dates.UTC), "yyyymmddTHHMMSSZ"))
```
so we probably should coordinate with @mattBrzezinski about names if we want to use a new one. (Maybe I should not have been running CI on this PR actually because of this-- sorry!)

Reply:
Ohhhh... sorry sorry, I forgot you are testing these on AWS, where this sort of thing matters more.  Apologies!  I only created the bucket to absolutely guarantee that there are no existing names in the bucket.  Regardless will fix!